src/solvers/refinement/string_refinement.cpp

```diff
@@ -1342,6 +1353,12 @@ static std::pair<bool, std::vector<exprt>> check_axioms(
   const auto eom=messaget::eom;
   static const std::string indent = "  ";
   static const std::string indent2 = "    ";
+  // clang-format off
+  const auto gen_symbol = [&](const irep_idt &id, const typet &type)
+  {
+    return generator.fresh_symbol(id, type);
+  };
```

Comment:
Why have the function as a parameter; why not just call generator.fresh_symbol?  Fresh symbol functions feel kind of stateful to me so I think I'd prefer them to be member functions.

Reply:
Also, can we change the description of the PR as this looks like a refactoring rather than a bug fix.

Reply:
@martin-cs @romainbrenguier is on holiday now so I'm taking over his open PRs. About `fresh_symbol`, it seems @romainbrenguier already adressed your comment. About the name of the PR, the main goal was to fix some bugs. The refactoring is just a side-effect of that and the commits messages are explicit enough. Do you agree?